src/Microsoft.Extensions.Diagnostics.HealthChecks/IHealthCheckService.cs

```diff
@@ -1,26 +1,38 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System.Collections.Generic;
+using System;
 using System.Threading;
 using System.Threading.Tasks;
 
 namespace Microsoft.Extensions.Diagnostics.HealthChecks
 {
     /// <summary>
-    /// A service which can be used to check the status of <see cref="IHealthCheck"/> instances registered in the application.
+    /// A service which can be used to check the status of <see cref="IHealthCheck"/> instances 
+    /// registered in the application.
     /// </summary>
+    /// <remarks>
+    /// <para>
+    /// The default implementation of <see cref="IHealthCheckService"/> is registered in the dependency
+    /// injection container as a singleton service by calling 
+    /// <see cref="DependencyInjection.HealthCheckServiceCollectionExtensions.AddHealthChecks(DependencyInjection.IServiceCollection)"/>.
+    /// </para>
+    /// <para>
+    /// The <see cref="IHealthChecksBuilder"/> returned by 
+    /// <see cref="DependencyInjection.HealthCheckServiceCollectionExtensions.AddHealthChecks(DependencyInjection.IServiceCollection)"/>
+    /// provides a convenience API for registering health checks.
+    /// </para>
+    /// <para>
+    /// The default implementation of <see cref="IHealthCheckService"/> will use all services
+    /// of type <see cref="IHealthCheck"/> registered in the dependency injection container. <see cref="IHealthCheck"/>
+    /// implementations may be registered with any service lifetime. The implementation will create a scope
+    /// for each aggregate health check operation and use the scope to resolve services. The scope
+    /// created for executing health checks is controlled by the health checks service and does not
+    /// share scoped services with any other scope in the application.
+    /// </para>
+    /// </remarks>
     public interface IHealthCheckService
     {
-        /// <summary>
-        /// A <see cref="IReadOnlyDictionary{TKey, T}"/> containing all the health checks registered in the application.
-        /// </summary>
-        /// <remarks>
-        /// The key maps to the <see cref="IHealthCheck.Name"/> property of the health check, and the value is the <see cref="IHealthCheck"/>
-        /// instance itself.
-        /// </remarks>
-        IReadOnlyDictionary<string, IHealthCheck> Checks { get; }
```

Comment:
This has to go away now because they aren't singletons. The main use case we had for providing this was to support filtering. If someone wants to get the set of checks for an arbitrary reason they can create a scope and resolve them 😆